### ISSUE TITLE
START.DAT size limitations

### ISSUE THREAD

**Issue by Kelebek1** [ProgSys/pg_disatools#21]
Reference issue #16

In the D1 exe (versions 1.0.0.1 and 1.0.0.5 have the same addresses), there's a function at disgaea.exe+0x317E0 which controls the memory allocation for files: https://i.imgur.com/Yn1tl3g.png

The highlighted values are the memory size that needs to increased, by default it's 0x15400000.

You can easily change it via hex editor, just by searching for that value in little endian. https://i.imgur.com/JatFgu6.png - Here are the 4 instances, and I've replaced them with 0x20000000, giving another ~180MB and preventing crashes with larger START.DAT files.

Also, if you could come help figure out the D5 sprite format that'd be great. <3

**Comment by ProgSys** [comment]
Nice work! That is quite a christmas present!

I implemented a small utility inside the File Manager to try it out and it seams to work so far:
<img src="https://user-images.githubusercontent.com/4257741/70159928-f85fa500-16b9-11ea-99b5-4dbc481a1d51.png" height="300">

The only thing left are some people who can make use out of it^^

I am not much motivated to work on D5, because they encrypted the files. Yes the encryption is very trivial, but someone took their time to add it in. This tells to me indirectly, that they don't want for people to play around with the files. In Disgaea PC they at least have a Workshop sub forum.

**Comment by Kelebek1** [comment]
@ProgSys It's extremely basic, and we have everything extracted and decrypted and re-buildable. I can edit and re-import sprites, I made this monster for example: https://i.imgur.com/OCiDZse.mp4 - from a sprite sheet someone made for D1. All the alignment is messed up and yeah it's weird, but it was a meme for the discord and a basic proof of concept. I just don't understand all the table values because I know nothing about sprites, and in particular I can't find cutout x/y, and I don't have the ability to make a nice sprite editor like yours.

I have full extraction and building of script.dat for custom scripting (https://i.imgur.com/B7JOOWP.mp4 and https://i.imgur.com/ixL9pe6.mp4), and the vast majority of files in Database_0 are mapped out and understood, I've made custom map edits as well (https://media.discordapp.net/attachments/534998804433469451/551474612970717202/Untitled-1.png), so map editors are possible. I've been mapping out the save file structure as well for save editing etc. Really everything's up for grabs.

Up to you though of course. :)
